### DISCUSSION TITLE
Selective room cleaning with Siri voice control

### DISCUSSION THREAD

**Discussion by pascalep98** [merdok/homebridge-miot#307] [Ideas]
Hey, first of all I love your plugin as it allowed me to automate my Dreame D9 Vacuum cleaner. Thank you!

My question/ idea is to add a function for selective room cleaning. In the xiaomi home app it is possible to select a specific room and let it clean but it does not work with Siri. I would imagine something like: "Hey Siri, clean the kitchen" after you cooked your dinner or so. Would be realle helpful in my opinion. 

I read a lot about it but it does not seem to work with your plugin yet. Here is a post of another plugin for roborock devices that I stumbled about. https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum

I must say that I'm a real noob regarding homebrige and coding in general but maybe you as the creator has an idea how to solve it. 

Thank you and have a great day :)

**Comment by merdok** [comment]
Thanks for the feedback!
Actually it is possible to do what you desire, you can create action buttons for every room which you like (or even group of rooms) and then you can use siri to trigger those buttons and start the desired room cleaning. More info on that here: https://github.com/merdok/homebridge-miot/blob/main/docs/robotcleaner.md#dreame

**Reply by pascalep98** [reply]
Hey thank you for the reply. I'll try to implement it as you suggested. Thank you :)

**Reply by hiSweid** [reply]
Hello, I can't get the room cleaning to run using the action buttons. I have a roborock s7 (roborock.vacuum.a15) on the metadata fetcher it says: 
vacuum:start-room-sweep | 2.6 (Start room sweep piid 9

so my action is: 

`"actionButtons": [
                        {
                            "action": "2.6",
                            "name": "Bath",
                            "params": [
                                "22"
                            ]
                        }
                    ]`

i got the param number "22" for my room id from another homebridge plugin named: Homebridge Xiaomi Roborock Vacuum and here everything is working.

When i start the action button, the robot just starts normal sweeping instead of room sweeping

what did i do wrong?

**Reply by merdok** [reply]
The params for room cleaning are usually far more complex then that (have a look at the guides for Dreame and Viomi). Somebody would need to use the modified mi home app on an android device to capture the commands which are sent to the device. That way we could know how the command (params) needs to look exactly.
